design/REGO_V2_PROPOSAL.md

```diff
@@ -184,44 +220,79 @@ cannot be reassigned.
 
 ## Decisions
 
-A decision is simply a return statement within a policy function. The types of
-all decisions must agree. For authorization policies, the decisions will
-typically be boolean with a default decision of return false to indicate deny
-by default semantics.
+A decision is simply a return statement within a rule function. The types of
```

Comment:
Can we refine the wording here? What do we mean by the types of all decisions must agree? I could imagine having two rules in the same package, one produces a map of strings and another produces a set of ints. Do those agree?

Reply:
I don't think those rules agree, but I see your point in that we should be more flexible about what kind of value is returned since the code driving policy execution may also expect a dynamically typed output.